analytics/analytics/analytic_unit_worker.py

```
@@ -35,11 +35,85 @@ def __init__(self, analytic_unit_id: str, detector: detectors.Detector, executor
             raise Exception('Timeout ({}s) exceeded while learning'.format(config.LEARNING_TIMEOUT))
 
     async def do_detect(self, data: pd.DataFrame, cache: Optional[ModelCache]) -> dict:
-        return self._detector.detect(data, cache)
+        if cache is None:
+            msg = f'{self.analytic_unit_id} detection got invalid cache, skip detection'
+            logger.error(msg)
+            raise ValueError(msg)
+        
+        #TODO choose chunk size for thresholds without window size
+        window_size = cache.get('WINDOW_SIZE', 1)
```

Let's raise an exception when there is no `WINDOW_SIZE` in `cache`